### ISSUE TITLE
hostname doesn't resolve to IP, even if I use openshift_hostname 

### ISSUE THREAD

**Issue by mjavadi** [openshift/openshift-ansible#2412]
Running ansible 2.1.1.0.
Using origin by doing a git clone of openshift/ansible on 9/6

I am using a hosts file for a single master per example on the origin docs.

The error is as follows (name and domain were altered prior to posing there).  The process does complete.  I tried adding openshift_hostname to the line on the master but getting the same results. I am trying to get router work and hit a test app/pod/svc.  Although I am able to start the svc and create a router, I am still not able to reach the app.  Probably a red herring and newbie error.  

Not sure what is causing the warning.  The hostname(s) match.  No entries have been added to the hosts file. That is the name of the VM in our environment. 

TASK [Warn user about bad openshift_hostname values] ***************************
[Warn user about bad openshift_hostname values]
The hostname "jcc-192-168-1-171.abc.acme.com" for "jcc-192-168-1-171.abc.acme.com" doesn't resolve to an ip address owned by this host. Please set openshift_hostname variable to a hostname that when resolved on the host in question resolves to an IP address matching an interface on this host. This host will fail liveness checks for pods utilizing hostPorts, press ENTER to continue or CTRL-C to abort.:

The hosts files on my ansible looks like below:
[OSEv3:children]
masters
nodes

[OSEv3:vars]
ansible_ssh_user=myuser
ansible_sudo=true
ansible_become=true
ansible_become_user=root
containerized=false
deployment_type=origin
openshift_router_selector='region=infra'
openshift_registry_selector='region=infra'
osm_default_subdomain='abc.acme.com'
openshift_http_proxy=80
openshift_https_proxy=443
openshift_generate_no_proxy_hosts=false

[masters]
jcc-128-149-53-171.abc.acme.com openshift_hostname='jcc-192-168-1.abc.acme.com'

[nodes]
jcc-192-168-1-171.abc.acme.com openshift_node_labels="{'region': 'infra', 'zone': 'jcc1'}" openshift_schedulable=false openshift_hostname='jcc-192-168-1-171.abc.acme.com'
jcc-192-168-1-172.abc.acme.com openshift_node_labels="{'region': 'primary', 'zone': 'jcc1'}"
jcc-192-168-1-173.abc.acme.com openshift_node_labels="{'region': 'primary', 'zone': 'jcc1'}"

**Comment by hzgraham** [comment]
Here is what I gathered from the code, someone will correct me if I'm wrong...The error is returned when "openshift_hostname" doesn't resolve to an IP in the "ansible_all_ipv4_addresses" fact for that node

It is determining that by running this command to get an IP which it will then check against that fact:
```
getent ahostsv4 {{ openshift.common.hostname }} | head -n 1 | awk '{ print $1 }'
```
or in your case
```
getent ahostsv4 jcc-192-168-1-171.abc.acme.com | head -n 1 | awk '{ print $1 }'
```
Does running that command return an IP used by that host?

**Comment by mjavadi** [comment]
This an Openstack vm...Have not tried this in AWS.  
I modified the /etc/hostfile and removed the line that has

127.0.01 jcc-192-168-1-171.abc.acme.com

Still got the same. Then I removed the lines that are ipv6 related. 
 The following lines are desirable for IPv6 capable hosts... At that point the command returned the real ip instead of 127.0.0.1
::1 jcc-192-168-1-171.abc.acme.com jcc-192-168-1-171
::1 localhost.localdomain localhost
::1 localhost6.localdomain6 localhost6
